### PR TITLE
Keyboard control feature

### PR DESCRIPTION
**Pull request by ktpa**

### CODE REVIEW

File: frontend/src/components/G6Graph.svelte [hunk 1]
```diff
@@ -21,13 +23,15 @@
         dispatch("nodeselected", null);
     };
 
+    // This is a hack to get the graph to render properly
```
Comment (RuthgerD) [inline frontend/src/components/G6Graph.svelte:26]: would prefer this to be a separate PR

File: frontend/src/components/G6Graph.svelte [hunk 2]
```diff
@@ -115,6 +119,19 @@
 
         graph.on("nodeselectchange", (e) => dispatch("nodeselected", e));
 
+        // Enable keyboard manipulation
+        graph.on("keydown", (e: IG6GraphEvent) => {
+            if (e.key === "ArrowRight") {
```
Comment (RuthgerD) [inline frontend/src/components/G6Graph.svelte:124]: Hate to be like this but I hate if chains with repeated calls more :)
```typescript
let weight = (k1, k2) => e.key == k1 ? -1 : e.key == k2 ? 1 : 0
graph.translate(
    weight("ArrowRight", "ArrowLeft") * graph_translation,
    weight("ArrowDown", "ArrowUp") * graph_translation
)
```

**Review by RuthgerD — CHANGES_REQUESTED** [review]
Would be nice if the transition was animated but thats a thing for later.

**Comment by ktpa** [comment]
> Would be nice if the transition was animated but thats a thing for later.

Resolved with animation_cfg.

**Review by RuthgerD — COMMENTED** [review]
Alright I tried it out locally and realized a problem with this approach, with these events you cant actually ever have 2 keys pressed at the same time; they will be separate events, causing jitter.

Its basically always locked into one dimension.
Do you think it is feasible to fix that?

**Comment by ktpa** [comment]
> Alright I tried it out locally and realized a problem with this approach, with these events you cant actually ever have 2 keys pressed at the same time; they will be separate events, causing jitter.
> 
> Its basically always locked into one dimension. Do you think it is feasible to fix that?

It may be feasible but as of now, the basic functionality is there and ItJustWorks. Multiple keydowns require some more tinkering and to configure the animation beyond the scope of the framework as it currently is. I've removed the animation due to the aforementioned stuttering issue.

**Review by RuthgerD — APPROVED** [review]
Thanks!

**Pull request merged**